doc/man/man3/crypto_x25519_dangerous_fast.3monocypher

```diff
@@ -0,0 +1,135 @@
+.\" A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+.\" HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+.\" SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+.\" LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+.\" DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+.\" THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+.\" (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+.\" OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+.\"
+.\" ----------------------------------------------------------------------------
+.\"
+.\" Written in 2020 by Fabio Scotoni
+.\"
+.\" To the extent possible under law, the author(s) have dedicated all copyright
+.\" and related neighboring rights to this software to the public domain
+.\" worldwide.  This software is distributed without any warranty.
+.\"
+.\" You should have received a copy of the CC0 Public Domain Dedication along
+.\" with this software.  If not, see
+.\" <https://creativecommons.org/publicdomain/zero/1.0/>
+.\"
+.Dd March 24, 2020
+.Dt CRYPTO_X25519_DANGEROUS_FAST 3monocypher
+.Os
+.Sh NAME
+.Nm crypto_x25519_dangerous_fast ,
+.Nm crypto_x25519_dangerous_small
+.Nd generation of Curve25519 points including low-order points
+.Sh SYNOPSIS
+.In monocypher.h
+.Ft void
+.Fo crypto_hidden_to_curve_fast
+.Fa "uint8_t pk[32]"
+.Fa "const uint8_t sk[32]"
+.Fc
+.Ft void
+.Fo crypto_hidden_to_curve_small
+.Fa "uint8_t pk[32]"
+.Fa "const uint8_t sk[32]"
+.Fc
+.Sh DESCRIPTION
+These functions are used in public key generation for
+.Xr crypto_curve_to_hidden 3monocypher .
+.Sy This is a highly advanced feature ;
+unless you are reading this because you were referred here from
+.Xr crypto_curve_to_hidden 3monocypher ,
+.Sy you likely have no reason to be using these functions
+and are probably looking for
+.Xr crypto_key_exchange 3monocypher
+or
+.Xr crypto_x25519 3monocypher
+instead.
+Expect elliptic curve jargon on this page.
+.Pp
+Both functions generate a Curve25519 public key
+.Fa pk
+from the given secret key
+.Fa sk ;
+the public keys are on the
+.Em whole
+curve, rather than just the main prime-order subgroup.
+but do the same with different code size and memory characteristics:
+.Fn crypto_x25519_dangerous_fast
+uses multiple large temporary variables and uses functions that are
+normally used internally for
+.Xr crypto_sign 3monocypher ;
+accordingly, it uses both more memory (for the temporary variables) and
+more code size (unless the signing code is already compiled in
+elsewhere).
+.Fn crypto_x25519_dangerous_small
+yields the same result, but does so using less time and memory at a
+large performance penalty compared to
+.Fn crypto_x25519_dangerous_fast .
+.Pp
+The resulting public keys
+.Em may
+be used with
+.Xr crypto_x25519 3monocypher
+and
+.Xr crypto_key_exchange 3monocypher ,
```

Comment:
Actually, that's your only choice. They are *intended* to be used with those, by design.